### PR TITLE
Production/master manifest generation in prow/release pipelines

### PR DESCRIPTION
**Pull request by MStokluska**
# Description
Jira: https://issues.redhat.com/browse/MGDAPI-304

- Edited the manifest/generate script to differentiate between production and master
- Enabled manifest checks which compare current master manifest on PR branch against a manifest generated on that branch during PR submission
- Enabled support for Delorean to generate a production manifest which matches current production release
- Added 2 initial manifest files to start off the automated process

# Verification
From this branch run
- `make manifest/rhoam` to generate `rhoam-master-manifest.txt` file
- `TYPE_OF_MANIFEST=production make manifest/rhoam` to generate production manifest based of current branch
- `TYPE_OF_MANIFEST=compare make manifest/rhoam` to compare current master manifest vs master manifest generated of this branch

Alternatively, for full verification, copy this branch, add a new package to integreatly project, push a new PR without the `make manifest/rhoam` - the manifest check on your PR will fail until you run the `make manifest/rhoam` command.

### CODE REVIEW

File: scripts/prodsec-manifest-generator.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# used. The generated manifest corresponds to a released version and is located in : ../rhoam-manifests/
+
+# Compares current manifest vs manifest generated for master
+manifest_compare() {
+    # Additional sorting of the master manifest is required as it seems that prow sorts files in a different manner.
+    sort -u "$RHOAM_CURRENT_MASTER" > "$RHOAM_SORTED_MASTER"
+
+    MANIFESTS_DIFF=$(diff --suppress-common-lines ${RHOAM_MASTER_FROM_BRANCH} ${RHOAM_SORTED_MASTER})
+    if [ ! -z "$MANIFESTS_DIFF" ]; then
+        echo "Difference found between master manifests, run `TYPE_OF_MANIFEST=master make manifest/prodsec` and push PR again"
```
Comment (Boomatang) [inline scripts/prodsec-manifest-generator.sh:13]: Can we remove the `"and push PR again"`, This command can be call in cases out side of a PR

File: scripts/prodsec-manifest-generator.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    case $OLM_TYPE in
+      "integreatly-operator")
+        OLM_TYPE="rhmi"
+        VERSION=$(grep integreatly-operator.v deploy/olm-catalog/integreatly-operator/integreatly-operator.package.yaml | tail -c 6)
+        FILE_NAME="rhmi-production-release-manifest.txt"
+        SERVICE_NAME="services-rhmi"
+        ;;
+      "managed-api-service")
+        OLM_TYPE="rhoam"
+        VERSION=$(grep managed-api-service.v deploy/olm-catalog/managed-api-service/managed-api-service.package.yaml | tail -c 6)
```
Comment (Boomatang) [inline scripts/prodsec-manifest-generator.sh:78]: This needs to change. the `tail -c 6` will give the wrong result if the version number was `managed-api-service.v1.11.0`. The string need to split on the `managed-api-service.v` and use the second element in the result.

File: scripts/prodsec-manifest-generator.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    manifest_generate
+    manifest_compare
+    exit 0
+    ;;
+"production")
+    case $OLM_TYPE in
+      "integreatly-operator")
+        OLM_TYPE="rhmi"
+        VERSION=$(grep integreatly-operator.v deploy/olm-catalog/integreatly-operator/integreatly-operator.package.yaml | tail -c 6)
```
Comment (Boomatang) [inline scripts/prodsec-manifest-generator.sh:72]: This needs to change. the `tail -c 6` will give the wrong result if the version number was `integreatly-operator.v2.11.0`. The string need to split on the `integreatly-operator.v` and use the second element in the result.

**Review by Boomatang — CHANGES_REQUESTED** [review]
There is one bug that needs to be addressed before this can be approved.

File: scripts/prodsec-manifest-generator.sh (hunk 2 above)
Comment (MStokluska) [reply scripts/prodsec-manifest-generator.sh]: done, thanks for review Jim!

File: scripts/prodsec-manifest-generator.sh (hunk 1 above)
Comment (MStokluska) [reply scripts/prodsec-manifest-generator.sh:13]: As agreed, resolving this conversation.

**Comment by MStokluska** [comment]
Thanks for the review @Boomatang, production logic was reviewed as part of release pipeline against my own repo. Master manifest creation can be reviewed by visiting the ci/prow/manifest link.

**Pull request merged**